src/payment/checkout/payment-form/flex-microform/CreditCardNumberField.jsx

```diff
@@ -0,0 +1,49 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { getCardIconForType } from '../utils/credit-card';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { FormattedMessage } from '@edx/frontend-platform/i18n';
+
+import FlexMicroformField from './FlexMicroformField';
+
+class CreditCardNumberField extends React.Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            cardIcon: null,
+        }
+        this.onNumberChange = this.onNumberChange.bind(this);
```

Comment:
FYI - you can avoid needing to remember `bind` lines like this by declaring the method as:

```
onNumberChange = (data) => {

};
```

The arrow function ensures that `this` is always bound to this scope.